src/main/kotlin/com/bantolomeus/service/BookService.kt

```diff
@@ -16,84 +16,87 @@ class BookService(private val bookRepository: BookRepository,
 
     fun createBook(bookDTO: BookDTO): BookDTO {
         val savedBookDTO = bookRepository.saveBookIfItNotExists(bookDTO)
-        if (savedBookDTO.currentPage > 0L) {
+        if (savedBookDTO.currentPage > 0) {
             saveBookUpdate(savedBookDTO)
         }
         return bookDTO
     }
 
     fun updateBook(bookUpdate: BookUpdateInputDTO, bookName: String): BookUpdatesFileDTO {
-        var response = BookUpdatesFileDTO()
-        var books = bookRepository.getBooks().books.filter { it.name != bookName }
-        var oldPage = Long.MAX_VALUE
-        val foundBook = bookRepository.getBookByName(bookName)
+        val foundBook = bookRepository.getBookByName(bookName) ?: return BookUpdatesFileDTO()
 
-        if (foundBook.name == bookName) {
-            foundBook.let {
-                oldPage = it.currentPage
-                it.currentPage = bookUpdate.currentPage
-                if (it.currentPage >= it.pagesTotal) {
-                    it.readTime = (Date().time - dateFormat.parse(it.dateStarted).time) / DIVISOR_FOR_DAY
-                }
-            }
-            books = mutableListOf(foundBook) + books.toMutableList()
+        if (foundBook.currentPage == foundBook.pagesTotal) {
```

Comment:
Why is there space before the if but not before the next if?

Reply:
same reason. Spaces are not about the ifs but about separating precondition checks from the business logic